sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -139,7 +203,8 @@ static status_t entropy_config(
   TRY(dif_entropy_src_configure(
       &entropy_src,
       (dif_entropy_src_config_t){
-          .fips_enable = true,
+          // FIXME: enable unconditionally when software can set FIPS mode.
+          .fips_enable = !entropy_insert_enable,
           .route_to_firmware = false,
           .bypass_conditioner = false,
```

Comment:
Can we also test the bypass mode? And what do you think about the route to SW?

Reply:
You know what, it's probably best to wait for @vogelpi to chip in here. I just read this:
https://github.com/lowRISC/opentitan/blob/050ca147cd71f13501cc4528fb37c2d4b9a8400a/hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson#L242
and thought that this might be something we want.
In a nutshell you would set route_to_firmware to true and then the final entropy that comes out of the entropy_src wouldn't be routed to hardware but to firmware and it would then be readable via [ENTROPY_DATA](https://opentitan.org/book/hw/ip/entropy_src/doc/registers.html#entropy_data).
If you want to bypass the conditioner at the same time you can but you'd have to set [es_route](https://opentitan.org/book/hw/ip/entropy_src/doc/registers.html#entropy_control--es_type) to true as well.
You should be able to find a more detailed explanation [here](https://opentitan.org/book/hw/ip/entropy_src/doc/programmers_guide.html#firmware-override--bypass-modes). (I also made some additions to this section which are not merged yet [here](https://github.com/lowRISC/opentitan/pull/21528))

In conclusion I'd at least test it with the bypass mode for now and wait for what @vogelpi has to say in regards to the route_to_firmware.

Reply:
We should test the bypass mode (conditioner off) but I'll comment in a different place regarding this.

We don't need to test the `.route_to_firmware` option here. This is a separate feature in the testplan and the main purpose of this feature (AFAIK) is to test the conditioner in isolation: i.e., you write a known value into the preconditioner FIFO, then let the conditioner process and read the produced output seed from ENTROPY_DATA. This is also known as known answer test (KAT) and it's what the existing test `chip_sw_entropy_src_known_answer_tests` does. Disabling the conditioner in this test would be possible but it's kind of the next step when debugging the test in order to find out if at least the path FW_OV_WR_DATA -> Bypass FIFO -> ENTROPY_DATA works. I think this is so basic that it doesn't require a test.